src/psyclone/tests/lfric_ref_elem_test.py

```diff
@@ -277,5 +280,166 @@ def test_union_refelem_gen(tmpdir):
             "f4_proxy%data, m3_proxy%data, m4_proxy%data, ndf_w1, undf_w1, "
             "map_w1(:,cell), ndf_w2, undf_w2, map_w2(:,cell), ndf_w3, undf_w3,"
             " map_w3(:,cell), nfaces_re_h, nfaces_re_v, "
-            "out_normals_to_horiz_faces, normals_to_vert_faces, "
+            "normals_to_vert_faces, out_normals_to_horiz_faces, "
             "out_normals_to_vert_faces)" in gen)
+
+
+def test_all_faces_refelem_gen(tmpdir):
+    ''' Test for code-generation for an invoke containing a single kernel
+    requiring all faces of reference-element (also check that only one
+    number of faces is passed to the kernel). '''
+    _, invoke_info = parse(os.path.join(BASE_PATH,
+                                        "23.4_ref_elem_all_faces_invoke.f90"),
+                           api=TEST_API)
+    psy = PSyFactory(TEST_API, distributed_memory=False).create(invoke_info)
+
+    assert LFRicBuild(tmpdir).code_compiles(psy)
+    gen = str(psy.gen).lower()
+
+    assert (
+        "      reference_element => mesh%get_reference_element()\n"
+        "      nfaces_re = reference_element%get_number_faces()\n"
+        "      call reference_element%get_normals_to_faces(normals_to_faces)\n"
+        "      call reference_element%get_outward_normals_to_faces("
+        "out_normals_to_faces)\n" in gen)
+    assert ("call testkern_ref_elem_all_faces_code(nlayers, a, f1_proxy%data, "
+            "f2_proxy%data, m1_proxy%data, m2_proxy%data, ndf_w1, undf_w1, "
+            "map_w1(:,cell), ndf_w2, undf_w2, map_w2(:,cell), ndf_w3, undf_w3,"
+            " map_w3(:,cell), nfaces_re, normals_to_faces, "
+            "out_normals_to_faces)" in gen)
+
+
+# Tests for generating the kernel stub code
```

Comment:
I've know we've not done this in other places (we will eventually!) but please could you break-out the kernel-stub tests into a separate file?

Reply:
Done. I am happy to move call and stub test into the `src/psyclone/tests/domain/lfric` directory in the next review round.
I also noticed that it was possible to define duplicate reference element metadat so I added a check in `class RefElementMetaData` and the `test_mdata_duplicate_var` test in `lfric_ref_elem_test.py`.

Reply:
Thanks for doing that. If you could just move the new file then that would be fine.

Reply:
Done with a small update to the `BASE_PATH`.